code/clean/clean_mi.do

```diff
@@ -66,25 +67,51 @@ replace settle = 1 if outcome == "P02 - Post-Charge Settlement Agreement"
-replace basis_clean = "Age" 		if basis == "Age"
-replace basis_clean = "Retaliation" if basis == "Retaliation"
-replace basis_clean = "Other" 		if regexm(basis, "Genetic") | basis == "Other" | basis == "Relationship/Assn." | basis == "Unassigned" | basis == "" // if it's missing
+// Clean basis (general)
+split basis, parse(;)
+
+foreach a of numlist 7/12 {
+	drop basis`a'
+}
+
+foreach a of numlist 1/6 {
+	replace basis`a' = "" if basis`a' == "?" | basis`a' == " ?" | strpos(basis`a', "?: ?") > 0
+	replace basis`a' = subinstr(basis`a', ": ", ":", .)
+	split basis`a', parse(:)
+	drop basis`a'
+}
+
+// SH
+g sh = 0
+replace sh = 1 if basis11 == "Sexual harassment" | (basis21 == "Sexual harassment" & basis11 == "") | (basis31 == "Sexual harassment" & basis11 == "" & basis21 == "")
+// no cases that are SH but not sex-based
+
+// basis_clean
+foreach a of numlist 1/6 {
+	drop basis`a'1
+	rename basis`a'2 basis`a'
+	replace basis`a' = subinstr(basis`a', ", ", ",", .)
+	split basis`a', parse(,)
+	drop basis`a'
+}
+
+g basis_clean = "Sex" 				if basis11 == "Sex" | (basis21 == "Sex" & basis11 == "") | (basis31 == "Sex" & basis11 == "" & basis21 == "")
+replace basis_clean = "Religion" 	if basis11 == "Religion"
+replace basis_clean = "Race"		if basis11 == "Race" | (basis21 == "Race" & basis11 == "")
+replace basis_clean = "Nationality" if basis11 == "National Origin" | (basis21 == "National Origin" & basis11 == "")
+replace basis_clean = "Disability"  if basis11 == "Disability" | (basis21 == "Disability" & basis11 == "")
+replace basis_clean = "Age" 		if basis11 == "Age" | (basis21 == "Age" & basis11 == "")
+replace basis_clean = "Retaliation" if basis11 == "Retaliation" | (basis21 == "Retaliation" & basis11 == "")
+replace basis_clean = "Other" 		if inlist(basis11, "Arrest Record", "Color", "Familial Status", "Height", "Marital Status", "Weight") | ///
+(basis21 == "Familial Status" & basis11 == "") | (basis11 == "" & basis21 == "" & basis31 == "")
+replace basis_clean = "Sex"         if sh == 1
+
+drop basis11-basis61
```

Comment:
Did you check that `basis_clean` has the same number of values as `basis`?

Reply:
Yes, both have 17,854 entries.